packages/types/src/multiDomain.ts

```diff
@@ -2,53 +2,19 @@ import type { ClerkOptions } from './clerk';
 
 type StringOrURLFnToString = string | ((url: URL) => string);
 
-/**
- * DomainOrProxyUrl supports the following cases
- * 1) none of them are set
- * 2) only proxyUrl is set
- * 3) isSatellite and proxy is set
- * 4) isSatellite and domain is set
- */
-export type MultiDomainAndOrProxy =
-  | {
-      isSatellite?: never;
-      proxyUrl?: never | StringOrURLFnToString;
-      domain?: never;
-    }
-  | {
-      isSatellite: Exclude<ClerkOptions['isSatellite'], undefined>;
-      proxyUrl?: never;
-      domain: StringOrURLFnToString;
-    }
-  | {
-      isSatellite: Exclude<ClerkOptions['isSatellite'], undefined>;
-      proxyUrl: StringOrURLFnToString;
-      domain?: never;
-    };
+export type MultiDomainAndOrProxy = {
+  isSatellite?: ClerkOptions['isSatellite'];
+  proxyUrl?: StringOrURLFnToString;
+  domain?: StringOrURLFnToString;
+};
```

Comment:
I feel that we are loosing some good dx here. Did you encounter any issues with these ? I believe these types were simplified enough especially after removing the discriminated unions for frontent api and publishable key.

Reply:
I believe the simplification is in favor of DX. Having said that, `StringOrURLFnToString` can be renamed to `StringOrFunction` and `BooleanOrFunction`.

Reply:
I reverted these (and simplified other) types so we can discuss more about this. However I believe that the DX benefits here are not as great as we'd like them to be. The types are becoming too complex in some cases, and the errors generated by misusing the props are not readable anymore.

I think I'd rather have some extra runtime checks in dev mode throwing readable errors, rather than overcomplicated discriminated union types that are hard to understand